grafana/resource_builtin_role_assignment_test.go

```diff
@@ -48,6 +48,147 @@ func TestAccBuiltInRoleAssignment(t *testing.T) {
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: testAccProviderFactories,
+		CheckDestroy:      testAccBuiltInRoleAssignmentCheckDestroy(&br),
+		Steps: []resource.TestStep{
+			{
+				PreConfig: func() {
+					err := prepareDefaultAssignments()
+					if err != nil {
+						t.Errorf("error when creating built-in role ssignments %s", err)
+					}
+				},
+				Config: builtInRoleAssignmentUpdatePreConfig,
+				Check: resource.ComposeTestCheckFunc(
+					testAccBuiltInRoleAssignmentCheckExists("grafana_builtin_role_assignment.test_builtin_assignment"),
+					resource.TestCheckResourceAttr(
+						"grafana_builtin_role_assignment.test_builtin_assignment", "builtin_role", "Viewer",
+					),
+					resource.TestCheckResourceAttr(
+						"grafana_builtin_role_assignment.test_builtin_assignment", "roles.#", "1",
+					),
+					resource.TestCheckResourceAttr(
+						"grafana_builtin_role_assignment.test_builtin_assignment", "roles.0.uid", "viewer_test",
+					),
+					resource.TestCheckResourceAttr(
+						"grafana_builtin_role_assignment.test_builtin_assignment", "roles.0.global", "true",
+					),
+				),
+			},
+			{
+				Config: builtInRoleAssignmentUpdateConfig,
+				Check: resource.ComposeTestCheckFunc(
+					testAccBuiltInRoleAssignmentCheckExists("grafana_builtin_role_assignment.test_builtin_assignment"),
+					resource.TestCheckResourceAttr(
+						"grafana_builtin_role_assignment.test_builtin_assignment", "builtin_role", "Viewer",
+					),
+					resource.TestCheckResourceAttr(
+						"grafana_builtin_role_assignment.test_builtin_assignment", "roles.#", "2",
+					),
+					resource.TestCheckResourceAttr(
+						"grafana_builtin_role_assignment.test_builtin_assignment", "roles.0.uid", "viewer_test",
+					),
+					resource.TestCheckResourceAttr(
+						"grafana_builtin_role_assignment.test_builtin_assignment", "roles.0.global", "true",
+					),
+					resource.TestCheckResourceAttr(
+						"grafana_builtin_role_assignment.test_builtin_assignment", "roles.1.uid", "viewer_test_2",
+					),
+					resource.TestCheckResourceAttr(
+						"grafana_builtin_role_assignment.test_builtin_assignment", "roles.1.global", "true",
+					),
+					testAccBuiltInRoleAssignmentWereNotDestroyed(),
+				),
+			},
+		},
+	})
+}
+
+func prepareDefaultAssignments() error {
+	client := testAccProvider.Meta().(*client).gapi
+	r1 := gapi.Role{
+		UID:     "reportviewer",
+		Version: 1,
+		Name:    "Test Report Viewer",
+		Global:  true,
+		Permissions: []gapi.Permission{
+			{
+				Action: "reports:read",
+				Scope:  "reports:*",
+			},
+			{
+				Action: "users:create",
+			},
+		},
+	}
+	r2 := gapi.Role{
+		UID:     "createuser",
+		Version: 1,
+		Name:    "Test Create User",
+		Global:  true,
+		Permissions: []gapi.Permission{
+			{
+				Action: "users:create",
+			},
+		},
+	}
+	role1, err := client.NewRole(r1)
+	if err != nil {
+		return fmt.Errorf("error creating role: %s", err)
```

Comment:
```suggestion
		return fmt.Errorf("error creating role: %w", err)
```

nit: just so error un/wrapping still succeeds